platform/darwin/src/MGLShape.h

```diff
@@ -86,6 +86,15 @@ MGL_EXPORT
  */
 @property (nonatomic, copy, nullable) NSString *subtitle;
 
+/**
+ A Boolean value indicating whether the annotation is enabled. This does
+ not impact style layers created with subclasses of `MGLShape`.
+ 
+ The default value of this property is `YES`. If the value of this property is
+ `NO`, the `MGLShape` object ignores touch events and cannot be selected.
+ */
+@property (nonatomic, getter=isEnabled) BOOL enabled;
+
```

Comment:
Discussion point for later: I wonder if this property could be part of a protocol that `MGLShape`, `MGLAnnotationImage`, `MGLAnnotationView` should conform to?

Reply:
That could be interesting. `MGLAnnotation` covers two of the three (but not `MGLAnnotationImage` as far as I know).

Reply:
It’s important to distinguish between data and presentation: MGLShape, MGLAnnotation, etc. are all about data storage, whereas MGLAnnotationView and MGLAnnotationImage are ways to render that data. The same distinction can be made between MGLSource and MGLStyleLayer.

MGLAnnotationView and MGLAnnotationImage both have a existing `enabled` properties. How will the property being added in this PR interact with the existing ones? What if an annotation’s `enabled` property disagrees with its view or image’s `enabled` property at the time the user taps it?

In https://github.com/mapbox/mapbox-gl-native/issues/10539#issuecomment-400071989, I suggested that `MGLAnnotationImage.enabled` property be deprecated, but to me a property on MGLShape is worse because an aspect of the data’s presentation is defined on the data itself. The problem with `MGLAnnotationImage.enabled` is not that it’s on the wrong kind of class but that there’s no way to change the annotation image’s interactivity after it appears on the map for the first time. `MGLAnnotationView.enabled` doesn’t suffer from this problem, because you can call `-[MGLMapView viewForAnnotation:]` at any time.